editor/src/clj/editor/code/data.clj

```diff
@@ -2123,20 +2199,16 @@
                       (not-any? (partial cursor-range-equals? cursor-range) visible-cursor-ranges)))
                visible-occurrences))))
 
-(defn replace-typed-chars [indent-level-pattern indent-string grammar lines cursor-ranges regions ^LayoutInfo layout replaced-char-count replacement-lines]
-  (assert (not (neg? ^long replaced-char-count)))
-  (let [splices (mapv (fn [^CursorRange cursor-range]
-                        (let [adjusted-cursor-range (adjust-cursor-range lines cursor-range)
-                              start (cursor-range-start adjusted-cursor-range)
-                              new-start-col (- (.col start) ^long replaced-char-count)
-                              new-start (->Cursor (.row start) new-start-col)
-                              end (cursor-range-end adjusted-cursor-range)]
-                          (assert (not (neg? new-start-col)))
-                          [(->CursorRange new-start end) replacement-lines]))
-                      cursor-ranges)]
-    (-> (splice lines regions splices)
-        (fix-indentation-after-splice indent-level-pattern indent-string grammar)
-        (update-document-width-after-splice layout))))
+(defn replace-typed-chars [indent-level-pattern
+                           indent-string
+                           grammar
+                           lines
+                           regions
+                           ^LayoutInfo layout
+                           splices]
```

Comment:
Initially, I split it because I started adding more arguments, but in the end, it was refactored to have fewer arguments. I'll unroll it to a single line.